packages/transaction-controller/src/utils/batch.ts

```diff
@@ -529,3 +574,88 @@ async function processTransactionWithHook(
     params: newParams,
   };
 }
+
+/**
+ * Requests approval for a transaction batch by interacting with the ApprovalController.
+ *
+ * @param txBatchMeta - Metadata for the transaction batch, including its ID and origin.
+ * @param messenger - The messenger instance used to communicate with the ApprovalController.
+ * @returns A promise that resolves to the result of adding the approval request.
+ */
+async function requestApproval(
+  txBatchMeta: TransactionBatchMeta,
+  messenger: TransactionControllerMessenger,
+): Promise<AddResult> {
+  const id = String(txBatchMeta.id);
+  const { origin } = txBatchMeta;
+  const type = 'transaction_batch';
+  const requestData = { txBatchId: id };
+
+  return (await messenger.call(
+    'ApprovalController:addRequest',
+    {
+      id,
+      origin: origin || ORIGIN_METAMASK,
+      requestData,
+      expectsResult: true,
+      type,
+    },
+    true,
+  )) as Promise<AddResult>;
+}
+
+/**
+ * Create a new batch metadata object.
+ *
+ * @param options - The options for creating a new batch metadata object.
+ * @param options.id - The ID of the transaction batch.
+ * @param options.chainId - The chain ID of the transaction batch.
+ * @param options.networkClientId - The network client ID of the transaction batch.
+ * @param options.transactions - The transactions in the batch.
+ * @param options.origin - The origin of the transaction batch.
+ * @returns A new TransactionBatchMeta object.
+ */
+function newBatchMetadata({
+  id,
+  chainId,
+  networkClientId,
+  transactions,
+  origin,
+}: TransactionBatchMeta): TransactionBatchMeta {
+  return {
+    id,
+    chainId,
+    networkClientId,
+    transactions,
+    origin,
+  };
+}
+
+/**
+ * Adds batch metadata to the transaction controller state.
+ *
+ * @param transactionBatchMeta - The transaction batch metadata to be added.
+ * @param update - The update function to modify the transaction controller state.
+ */
+function addBatchMetadata(
+  transactionBatchMeta: TransactionBatchMeta,
+  update: UpdateBatchMetadata,
+) {
+  update((state: WritableDraft<TransactionControllerState>) => {
```

Comment:
Minor, as the callback is typed, can we omit the type of `state` in every usage?